crates/relayer/src/chain/ckb4ibc.rs

```diff
@@ -624,119 +670,82 @@ impl ChainEndpoint for Ckb4IbcChain {
         &mut self,
         tracked_msgs: TrackedMsgs,
     ) -> Result<Vec<IbcEventWithHeight>, Error> {
```

Comment:
I think [fn send_messages_and_wait_check_tx](https://github.com/synapseweb3/forcerelay/blob/81d2a863c0bd9b3e52eebfe7656c3eb5ce71197e/crates/relayer/src/chain/ckb4ibc.rs#L755-L774) should not call `fn send_messages_and_wait_commit`.

Because `send_messages_and_wait_commit` sends one or more transactions with `msgs` to chain and synchronously wait for it to be committed, while `send_messages_and_wait_check_tx` should be a **Non-blocking** alternative to `send_messages_and_wait_commit` interface.

See [Send transactions interfaces](https://github.com/synapseweb3/forcerelay/blob/81d2a863c0bd9b3e52eebfe7656c3eb5ce71197e/crates/relayer/src/chain/endpoint.rs#L139-L153)

Reply:
that's true, but I have no idea wether we should do as you said in Axon and CKB scenario because of the security issue, actually, we don't meet any problems so far, but I guess this works for Cosmos, and to fetch the interface's original meaning, I'll try it and check the function

Reply:
I see.
If you think it won't cause problems, please consider leaving a `TODO` comment for [fn send_messages_and_wait_check_tx](https://github.com/synapseweb3/forcerelay/blob/81d2a863c0bd9b3e52eebfe7656c3eb5ce71197e/crates/relayer/src/chain/ckb4ibc.rs#L755-L774) to later refactoring.